File: backend/src/main/java/at/ac/tuwien/ase/groupphase/backend/dto/Registration.java
package at.ac.tuwien.ase.groupphase.backend.dto;

import at.ac.tuwien.ase.groupphase.backend.entity.Region;
import lombok.Data;
import lombok.EqualsAndHashCode;

public record Registration(String email, String username, String password, Region region, String postalCode) {

}

File: frontend/src/app/dtos/registration.ts
import {Region} from "../shared/region";

export class Registration {

    /*email: string;
    username: string;
    password: string;
    region: Region;
    postalCode: string;*/
    constructor(
        public email: string,
        public username: string,
        public password: string,
        public region: Region,
        public postalCode: string
    ) {

    }
    toString(): string {
        return `object of type Registration:
         email: ${this.email},
         username: ${this.username},
         password: ${this.password},
         region: ${this.region.toString()},
         postalCode: ${this.postalCode},
         `;
    }
}

